rocky/rocky/views/mixins.py

```diff
@@ -82,26 +82,17 @@ class ObservedAtMixin:
 
     @cached_property
     def observed_at(self) -> datetime:
-        observed_at = self.request.GET.get("observed_at", None)
-        if not observed_at:
-            return datetime.now(timezone.utc)
+        now = datetime.now(timezone.utc)
+        observed_at_string = self.request.GET.get("observed_at", now.isoformat())
 
         try:
-            datetime_format = "%Y-%m-%d"
-            date_time = convert_date_to_datetime(datetime.strptime(observed_at, datetime_format))
-            if date_time.date() > datetime.now(timezone.utc).date():
-                messages.warning(self.request, _("The selected date is in the future."))
-            return date_time
-        except ValueError:
-            try:
-                ret = datetime.fromisoformat(observed_at)
-                if not ret.tzinfo:
-                    ret = ret.replace(tzinfo=timezone.utc)
-
-                return ret
-            except ValueError:
-                messages.error(self.request, _("Can not parse date, falling back to show current date."))
-                return datetime.now(timezone.utc)
+            observed_at = datetime.fromisoformat(observed_at_string).replace(tzinfo=timezone.utc)
```

Comment:
A minor optimization, but the problem here is in the case of a missing request parameter, you already have a ISO formatted UTC timestamp (due to line 86). Reparsing the already formatted string seems redundant. Another way to do this is (in pseudocode):
```python3
if "observed_at" in self.request.GET:
    observed_at = datetime.fromisoformat(self.request.GET['observed_at']).replace(tzinfo=timezone.utc)
else:
    observed_at = datetime.now(timezone.utc)
```